proto/buf/alpha/registry/v1alpha1/repository_branch.proto

```diff
@@ -67,6 +87,27 @@ message ListRepositoryBranchesResponse {
   string next_page_token = 2;
 }
 
+message ListRepositoryBranchesByReferenceRequest {
+  // The ID of the repository whose branches should be listed.
+  string repository_id = 1;
+  uint32 page_size = 2;
+  // The first page is returned if this is empty.
+  string page_token = 3;
+  // The reference by which to filter the returned branches.
+  oneof reference {
```

Comment:
I explicitly didn't make this `string reference` because we need a guarantee that the reference we are requesting is a VCSCommit